### PR TITLE
move OpProtoAndCheckerMaker from operator to op_proto_maker

### PR DESCRIPTION
**Pull request by jacquesqiao**
fix: https://github.com/PaddlePaddle/Paddle/issues/4219

### CODE REVIEW

File: paddle/framework/op_proto_maker.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+namespace paddle {
+namespace framework {
+
+// this class not only make proto but also init attribute checkers.
+class OpProtoAndCheckerMaker {
+ public:
+  OpProtoAndCheckerMaker(OpProto* proto, OpAttrChecker* op_checker)
+      : proto_(proto), op_checker_(op_checker) {}
+
+  ~OpProtoAndCheckerMaker() {
```
Comment (reyoung) [inline paddle/framework/op_proto_maker.h]: virtual `~OpProtoAndCheckerMaker`

File: paddle/framework/op_proto_maker_test.cc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "gtest/gtest.h"
+
+class TestAttrProtoMaker : public paddle::framework::OpProtoAndCheckerMaker {
+ public:
+  TestAttrProtoMaker(paddle::framework::OpProto* proto,
+                     paddle::framework::OpAttrChecker* op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddAttr<float>("scale", "scale of test op");
+    AddAttr<float>("scale", "scale of test op");
```
Comment (reyoung) [inline paddle/framework/op_proto_maker_test.cc]: ???

File: paddle/operators/prelu_op.h [hunk 3]
```diff
@@ -94,7 +94,7 @@ class PReluGradKernel : public framework::OpKernel {
     Transform(context.device_context(), out_ptr, out_ptr + numel, dout_ptr,
               dx_ptr, PReluGradFunctor<T>(alpha_ptr));
 
-    // TODO (Zhuoyuan): add dalpha upgrade when GPU kernels ready
+    // TODO(Zhuoyuan): add dalpha upgrade when GPU kernels ready
```
Comment (reyoung) [inline paddle/operators/prelu_op.h]: Maybe update develop branch.

File: paddle/framework/op_proto_maker.h (hunk 1 above)
Comment (jacquesqiao) [reply paddle/framework/op_proto_maker.h]: done

**Pull request merged**